### PR TITLE
Remove unnecessary spec, build_files, rpmmacro and package name attribs from package yaml

### PR DESCRIPTION
**Pull request by alanoe**

### CODE REVIEW

**Comment by alanoe** [comment]
I have based this on top of ae/remove_unnecessary_expects_source_attrib_from_package_yaml branch, that's why the commits of that branches show up here. They'll disappear from the list when that branch is merged.

**Comment by olavphibm** [comment]
Mention the macros file and package name in the PR name.
This also needs a rebase.

**Comment by alanoe** [comment]
Rebase done, PR description updated

**Comment by alanoe** [comment]
start build

**Comment by mopsfelder** [comment]
Are the removed parameters optional or inexistent?  That would be clearer if stated in the commit message why they are being removed.

Perhaps it should be a good idea to document the optional ones in open-power-host-os/versions#106.

**Comment by alanoe** [comment]
MO> Are the removed parameters optional or inexistent? That would be clearer if stated in the commit message why they are being removed.

Optional, although I don't see a reason why someone would want to define them. I'd remove them completely when we can break backwards compatibility with older package metadata yaml files. I've made it clear these parameters are now optional in commit msg.

MO> Perhaps it should be a good idea to document all of the optional ones in open-power-host-os/versions#106, like, for example, name is optional and derived from <package>.yaml filename.

Agreed and done.

File: golang-github-shurcooL-sanitized_anchor_name/golang-github-shurcooL-sanitized_anchor_name.yaml [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
  commit_id: '1dba4b3954bc059efc3991ec364f9f9a35f597d2'
  branch: 'master'
  expects_source: 'golang-github-shurcooL-sanitized_anchor_name'
  files:
   CentOS:
    '7':
+<<<<<<< HEAD
     spec: 'CentOS/7/golang-github-shurcooL-sanitized_anchor_name.spec'
+=======
+>>>>>>> Remove unnecessary RPM package specification file path attrib from packages metadata files
```
Comment (olavphibm) [inline golang-github-shurcooL-sanitized_anchor_name/golang-github-shurcooL-sanitized_anchor_name.yaml]: Remove those diff marks.

Comment (alanoe) [inline golang-github-shurcooL-sanitized_anchor_name/golang-github-shurcooL-sanitized_anchor_name.yaml]: Thanks for pointing out. This is an accidental result of so many rebases I had to do in this patch. Could we pls merge ASAP?

File: qemu/qemu.yaml [hunk 2]
```diff
@@ -8,6 +7,3 @@ Package:
  files:
   CentOS:
    '7':
```
Comment (alanoe) [inline qemu/qemu.yaml]: Agreed. Done

**Comment by olavphibm** [comment]
start build

**Comment by olavphibm** [comment]
start build

**Comment by olavphibm** [comment]
start build

**Comment by olavphibm** [comment]
start build

**Comment by olavphibm** [comment]
start build

**Pull request merged**